netket/exact.py

```diff
@@ -171,6 +171,45 @@ def full_ed(operator, first_n=1, compute_eigenvectors=False):
 ExactTimePropagation.iter = _ExactTimePropagation_iter
 
 
+def steadystate(lindblad, sparse=False, method="ed"):
+    r"""Computes `first_n` smallest eigenvalues and, optionally, eigenvectors
+    of a Hermitian operator by full diagonalization.
+
+    Args:
+        lindblad: The lindbladian encoding the master equation.
+        sparse: Whever to perform the diagonalization using sparse matrices.
+
+    """
+    from numpy import sqrt, matrix
+
+    if method == "ed":
+        if not sparse:
+            from numpy.linalg import eigh
+
+            lind_mat = matrix(lindblad.to_dense())
+
+            ldagl = lind_mat.H * lind_mat
+            w, v = eigh(ldagl)
+
+        else:
+            from scipy.sparse.linalg import eigsh
+
+            lind_mat = lindblad.to_sparse()
+            ldagl = lind_mat.H * lind_mat
```

Comment:
isn't there a way to do ED without forming this dense matrix?

Reply:
You can compute the spectrum of `L`, but it's non hermitian so it often comes with big errors.
The alternative would be an iterative method solving L\rho=0, which I plan to implement at some point